source/PrintData.cpp

```diff
@@ -0,0 +1,497 @@
+			variants = true;
+		else if(arg == "--sales")
+			sales = true;
+		else if(arg == "--loaded")
+			loaded = true;
+	}
+
+	if(sales)
+		PrintShipShipyards();
+	else if(loaded)
+		PrintLoadedShipStats(variants);
+	else
+		PrintBaseShipStats();
+}
+
+
+
+void PrintData::PrintBaseShipStats()
+{
+	cout << "model" << ',' << "category" << ',' << "chassis cost" << ',' << "loaded cost" << ',' << "shields" << ','
+		<< "hull" << ',' << "mass" << ',' << "drag" << ',' << "heat dis" << ','
+		<< "crew" << ',' << "bunks" << ',' << "cargo" << ',' << "fuel" << ','
+		<< "outfit" << ',' << "weapon" << ',' << "engine" << ',' << "gun mounts" << ','
+		<< "turret mounts" << ',' << "fighter bays" << ',' << "drone bays" << '\n';
+	for(auto &it : GameData::Ships())
+	{
+		// Skip variants and unnamed / partially-defined ships.
+		if(it.second.ModelName() != it.first)
+			continue;
+
+		const Ship &ship = it.second;
+		cout << it.first << ',';
+
+		const Outfit &attributes = ship.BaseAttributes();
+		cout << attributes.Category() << ',';
+		cout << ship.ChassisCost() << ',';
+		cout << ship.Cost() << ',';
+
+		auto mass = attributes.Mass() ? attributes.Mass() : 1.;
+		cout << attributes.Get("shields") << ',';
+		cout << attributes.Get("hull") << ',';
+		cout << mass << ',';
+		cout << attributes.Get("drag") << ',';
+		cout << ship.HeatDissipation() * 1000.<< ',';
+		cout << attributes.Get("required crew") << ',';
+		cout << attributes.Get("bunks") << ',';
+		cout << attributes.Get("cargo space") << ',';
+		cout << attributes.Get("fuel capacity") << ',';
+
+		cout << ship.BaseAttributes().Get("outfit space") << ',';
+		cout << ship.BaseAttributes().Get("weapon capacity") << ',';
+		cout << ship.BaseAttributes().Get("engine capacity") << ',';
+
+		int numTurrets = 0;
+		int numGuns = 0;
+		for(auto &hardpoint : ship.Weapons())
+		{
+			if(hardpoint.IsTurret())
+				++numTurrets;
+			else
+				++numGuns;
+		}
+		cout << numGuns << ',' << numTurrets << ',';
+
+		int numFighters = ship.BaysTotal("Fighter");
+		int numDrones = ship.BaysTotal("Drone");
+		cout << numFighters << ',' << numDrones << '\n';
+	}
+}
+
+
+
+void PrintData::PrintShipShipyards()
+{
+	cout << "ship" << ',' << "shipyards" << '\n';
+	map<string, set<string>> ships;
+	for(auto &it : GameData::Shipyards())
+	{
+		for(auto &it2 : it.second)
+		{
+			ships[it2->ModelName()].insert(it.first);
+		}
+	}
+	for(auto &it : GameData::Ships())
+	{
+		if(it.first != it.second.ModelName())
+			continue;
+		cout << it.first;
+		for(auto &it2 : ships[it.first])
+		{
+			cout << ',' << it2;
+		}
+		cout << '\n';
+	}
+}
+
+
+
+void PrintData::PrintLoadedShipStats(bool variants)
+{
+	cout << "model" << ',' << "category" << ',' << "cost" << ',' << "shields" << ','
+		<< "hull" << ',' << "mass" << ',' << "crew" << ',' << "cargo" << ','
+		<< "bunks" << ',' << "fuel" << ',' << "outfit" << ',' << "weapon" << ','
+		<< "engine" << ',' << "speed" << ',' << "accel" << ',' << "turn" << ','
```

Comment:
```suggestion
		<< "hull" << ',' << "mass" << ',' << "required crew" << ',' << "cargo" << ','
		<< "bunks" << ',' << "fuel" << ',' << "outfit space" << ',' << "weapon capacity" << ','
		<< "engine capacity" << ',' << "speed" << ',' << "accel" << ',' << "turn" << ','
```
Isnt this whole thing pretty much the same as above btw? Could be good to group them up in a big string thats like in a namespace here
We could even do one better, and have just one method of the other, and depending on which mode we're on we stop in the middle of this string or not, and we use attributes or baseAttributes; and obviously stop later on in the data printing

Reply:
Not quite the same but I'll look into grouping the common aspects.